dev/test/ignore-undefined.ts

```diff
@@ -0,0 +1,297 @@
+      }
+    );
+  });
+
+  it('in query cursors', () => {
+    const overrides: ApiOverride = {
+      runQuery: request => {
+        queryEquals(
+          request,
+          orderBy('foo', 'ASCENDING'),
+          startAt(true, FOO_MAP)
+        );
+        return stream();
+      },
+    };
+
+    return createInstance(overrides, {ignoreUndefinedProperties: true}).then(
+      firestore => {
+        return firestore
+          .collection('collectionId')
+          .orderBy('foo')
+          .startAt({bar: 'bar', baz: undefined})
+          .get();
+      }
+    );
+  });
+});
+
+describe('rejects undefined values', () => {
+  describe('in top-level call', () => {
+    it('to set()', () => {
+      return createInstance({}, {ignoreUndefinedProperties: true}).then(
+        firestore => {
+          expect(() => {
+            firestore
+              .doc('collectionId/documentId')
+              .set(undefined as InvalidApiUsage);
+          }).to.throw(
+            'Value for argument "data" is not a valid Firestore document. Input is not a plain JavaScript object.'
+          );
+        }
+      );
+    });
+
+    it('to create()', () => {
+      return createInstance({}, {ignoreUndefinedProperties: true}).then(
+        firestore => {
+          expect(() => {
+            firestore
+              .doc('collectionId/documentId')
+              .create(undefined as InvalidApiUsage);
+          }).to.throw(
+            'Value for argument "data" is not a valid Firestore document. Input is not a plain JavaScript object.'
+          );
+        }
+      );
+    });
+
+    it('to update()', () => {
+      return createInstance({}, {ignoreUndefinedProperties: true}).then(
+        firestore => {
+          expect(() => {
+            firestore.doc('collectionId/documentId').update('foo', undefined);
+          }).to.throw(
+            '"undefined" values are only ignored in object properties.'
+          );
+        }
+      );
+    });
+
+    it('to Query.where()', () => {
+      return createInstance({}, {ignoreUndefinedProperties: true}).then(
+        firestore => {
+          expect(() => {
+            firestore
+              .doc('collectionId/documentId')
+              .collection('collectionId')
+              .where('foo', '==', undefined);
+          }).to.throw(
+            '"undefined" values are only ignored in object properties.'
+          );
+        }
+      );
+    });
+
+    it('to Query.startAt()', () => {
+      return createInstance({}, {ignoreUndefinedProperties: true}).then(
+        firestore => {
+          expect(() => {
+            firestore
+              .doc('collectionId/documentId')
+              .collection('collectionId')
+              .orderBy('foo')
+              .startAt(undefined);
+          }).to.throw(
+            '"undefined" values are only ignored in object properties.'
+          );
+        }
+      );
+    });
+  });
+
+  describe('when setting is disabled', () => {
```

Comment:
`Reject undefined fields when setting is disabled`?

Reply:
The full test names are: 
"rejects undefined values when setting is disabled in set()"
"rejects undefined values when setting is disabled in create()"
...

The name starts with the first "define" in line 158.